### DISCUSSION TITLE
Ids from mei to svg

### DISCUSSION THREAD

**Discussion by samehchafik** [rism-digital/verovio#3780] [Q&A]
Hello to the whole team,

First of all, I want to express my gratitude for this software. You've done an excellent job, it’s a very useful tool.

I’m using Verovio as part of the development of an automatic music score recognition software. The engine provides me with a MEI file, which I need to implement to present to users so they can report errors generated during the recognition process.

The MEI file has IDs on all the important tags: clef, key signature, time signature, grpSym... and notes.

Last week, I ran some tests, and I was sure that you were preserving the IDs on the clef, key signature, time signature, etc. However, this week I realized that only the note IDs are preserved.

For example:

<clef xml:id="clef_1223_1710" shape="G" line="2"/>
<keySig xml:id="ks_1323_1721" sig="2f"/>
<meterSig xml:id="tsign24" count="4" unit="4"/>

The IDs clef_1223_1710, ks_1323_1721, and tsign24 appear in the SVG with random identifiers.

Could you please tell me how I can preserve these IDs? Is there an option I need to pass during initialization?

I am using Verovio in its JavaScript version in a Vue.js project.

Thank you in advance for any help.

**Comment by ahankinson** [comment]
Converted to a discussion since this is a support question and isn't an issue with the code.
